### ISSUE TITLE
yolov5m-cls.yaml

### ISSUE THREAD

**Issue by yu1206** [ultralytics/yolov5#13644]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

yolov5m-cls.yaml    is  where

### Additional

yolov5m-cls.yaml    is  where

**Comment by UltralyticsAssistant** [comment]
👋 Hello @yu1206, thank you for reaching out and for your interest in YOLOv5 🚀! This is an automated response to help you get started—an Ultralytics engineer will also review and assist you soon.

Please visit our ⭐️ [Tutorials](https://docs.ultralytics.com/yolov5/) for comprehensive guides, including [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) and other advanced topics.

If this is a 🐛 Bug Report, please provide a **minimum reproducible example** to help us investigate the issue.

If your question is about custom training or configuration files, please provide as much detail as possible, such as dataset examples, logs, and confirm you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

## Requirements
[**Python>=3.8.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed, including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```
## Environments
YOLOv5 can be run in these verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>
If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py) and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your patience and contribution! 🥳

**Comment by Y-T-G** [comment]
YOLOv5 classification models use the same YAML as detection models using the first 10 layers and then adding `Classify` head.
https://github.com/ultralytics/yolov5/blob/567c66463e943b731e08c9a9476660c13408f088/models/yolo.py#L361-L365

**Comment by liuxuet** [comment]
> YOLOv5 classification models use the same YAML as detection models using the first 10 layers and then adding `Classify` head.
> 
> [yolov5/models/yolo.py](https://github.com/ultralytics/yolov5/blob/567c66463e943b731e08c9a9476660c13408f088/models/yolo.py#L361-L365)
> 
> Lines 361 to 365 in [567c664](/ultralytics/yolov5/commit/567c66463e943b731e08c9a9476660c13408f088)
> 
>  model.model = model.model[:cutoff]  # backbone 
>  m = model.model[-1]  # last layer 
>  ch = m.conv.in_channels if hasattr(m, "conv") else m.cv1.conv.in_channels  # ch into module 
>  c = Classify(ch, nc)  # Classify() 
>  c.i, c.f, c.type = m.i, m.f, "models.common.Classify"  # index, from, type

So if I want to modify the model structure of the classification network, do I have to first modify the detection model, train it to obtain the model file, and then use this modified model file to pretrain the classification model?

**Comment by Y-T-G** [comment]
No. You just modify the classification model directly.

**Comment by liuxuet** [comment]
> No. You just modify the classification model directly.

class ClassificationModel(BaseModel):
    # YOLOv5 classification model
    def __init__(self, cfg=None, model=None, nc=1000, cutoff=10):  # yaml, model, number of classes, cutoff index
        super().__init__()
        self._from_detection_model(model, nc, cutoff) if model is not None else self._from_yaml(cfg)

    def _from_detection_model(self, model, nc=1000, cutoff=10):
        # Create a YOLOv5 classification model from a YOLOv5 detection model
        if isinstance(model, DetectMultiBackend):
            model = model.model  # unwrap DetectMultiBackend
        model.model = model.model[:cutoff]  # backbone
        m = model.model[-1]  # last layer
        ch = m.conv.in_channels if hasattr(m, 'conv') else m.cv1.conv.in_channels  # ch into module
        c = Classify(ch, nc)  # Classify()
        c.i, c.f, c.type = m.i, m.f, 'models.common.Classify'  # index, from, type
        model.model[-1] = c  # replace
        self.model = model.model
        self.stride = model.stride
        self.save = []
        self.nc = nc

    def _from_yaml(self, cfg):
        # Create a YOLOv5 classification model from a *.yaml file
        self.model = None


But in the ClassificationModel class, the classification model is converted from the passed-in model, and its model structure is the same as that of the detection model.

**Comment by Y-T-G** [comment]
Yes, but you don't need to pretrain then detection model
